### ISSUE TITLE
@openid/appauth integration not working when migrating to Quasar v2 from Quasar v1

### ISSUE THREAD

**Issue by vijay-singh-16** [quasarframework/quasar#11793]
Hi! fellow devs. We are using awesome Quasar v1 with capacitor mode to develop Android and IOS apps. Everything is working well. Recently we started migration to Quasar v2. Most things are working as expected with little tweaks here and there but we are not able to make the [@openid/appauth](https://github.com/openid/AppAuth-JS) to work.

Initially it was showing these build errors:

> App •  DONE  • "UI" compiled with errors • 27009ms
> 
> App •  ERROR  •  UI  in ./node_modules/@openid/appauth/built/node_support/crypto_utils.js        
> 
> Module not found: Can't resolve imported dependency "crypto"
> Did you forget to install it? You can run: yarn add crypto
> 
> App •  ERROR  •  UI  in ./node_modules/@openid/appauth/built/node_support/node_request_handler.js
> 
> Module not found: Can't resolve imported dependency "http"
> Did you forget to install it? You can run: yarn add http
> 
> App •  ERROR  •  UI  in ./node_modules/follow-redirects/index.js
> 
> Module not found: Can't resolve imported dependency "http"
> Did you forget to install it? You can run: yarn add http
> 
> App •  ERROR  •  UI  in ./node_modules/follow-redirects/index.js
> 
> Module not found: Can't resolve imported dependency "https"
> Did you forget to install it? You can run: yarn add https
> 
> App •  ERROR  •  UI  in ./node_modules/follow-redirects/index.js
> 
> Module not found: Can't resolve imported dependency "stream"
> Did you forget to install it? You can run: yarn add stream
> 
> App •  ERROR  •  UI  in ./node_modules/opener/lib/opener.js
> 
> Module not found: Can't resolve imported dependency "child_process"
> Did you forget to install it? You can run: yarn add child_process
> 
> App •  ERROR  •  UI  in ./node_modules/opener/lib/opener.js
> 
> Module not found: Can't resolve imported dependency "os"
> Did you forget to install it? You can run: yarn add os
> 
> App •  ERROR  •  UI  in ./src/auth/CapacitorCrypto.ts
> 
> Module not found: Can't resolve imported dependency "crypto"
> Did you forget to install it? You can run: yarn add crypto
> 
> 
> App •  COMPILATION FAILED  • Please check the log above for details.

most of which got fixed by [this tip](https://quasar.dev/start/upgrade-guide#nodejs-polyfills):

```
chainWebpack(chain) {
  const nodePolyfillWebpackPlugin = require('node-polyfill-webpack-plugin');
  chain.plugin('node-polyfill').use(nodePolyfillWebpackPlugin);
}
```
      
but now it shows this build error which we are not able to resolve:

> App •  ERROR  •  UI  in ./node_modules/opener/lib/opener.js
> 
> Module not found: Can't resolve imported dependency "child_process"
> Did you forget to install it? You can run: yarn add child_process

 We have already spent hours on this problem.

Please help!

**Comment by MilosPaunovic** [comment]
Have you tried the default solution - to remove `node_modules` and `package-lock.json` and do a fresh dependency install?

**Comment by vijay-singh-16** [comment]
Yes, we have tried this.
Even we tried to create fresh quasar v2 project and only added authentication module (having  @openid/appauth dependency)  to it, still we get the same build error.

**Comment by TobyMosque** [comment]
The new electron versions disable the node modules in the renderer process, in this case u'd two options:

1 - enable the nodeintegration (not recommended).
https://www.electronjs.org/docs/latest/tutorial/security#2-do-not-enable-nodejs-integration-for-remote-content

2 - move any service who relies in node modules to the preload script
https://quasar.dev/quasar-cli/developing-electron-apps/electron-preload-script

If u aren't on electron, pls refer the webpack 5 node polyfill:
https://webpack.js.org/configuration/node/

in both cases, the node integration/polyfill is disabled/not configured for security reasons.

**Comment by rstoenescu** [comment]
Toby is spot on. Approach no. 2 is recommended. This is a change in behavior done by the Electron team, and we're just following their recommendations.

**Comment by vijay-singh-16** [comment]
Thanks @TobyMosque and @rstoenescu.
Your replies made us try some other options which finally got the app working again with Quasar v2. Though we will be doing more testing of the app but I want to share the thing we currently opted to make the app work.

In the quasar.conf.js we added

`chain.resolve.alias.set('child_process', false);`

like this:

```
chainWebpack(chain) {
	const nodePolyfillWebpackPlugin = require('node-polyfill-webpack-plugin');
	chain.plugin('node-polyfill').use(nodePolyfillWebpackPlugin);
	
	chain.resolve.alias.set('child_process', false);// Added this
},
```

@TobyMosque, currently we are not targeting desktop, so I think electron option you mentioned in your reply is currently not required.
I will post an update in this thread, with details, whenever we target desktop.

**Comment by vijay-singh-16** [comment]
We have found the real reason of the build errors reported in the main question.
We were using a class 'NodeRequestor' from '@openid/appauth'. This class had many node based dependencies like 'http' and 'https' which further had other node based dependencies. When we replaced 'NodeRequestor' by another class 'FetchRequestor' from '@openid/appauth', all the errors went away, even when we removed this part from 'quasar.conf.js':

```
chainWebpack(chain) {
    const nodePolyfillWebpackPlugin = require('node-polyfill-webpack-plugin');
    chain.plugin('node-polyfill').use(nodePolyfillWebpackPlugin);
    
    chain.resolve.alias.set('child_process', false);
},
```

there were no errors.

Hope this may help some other devs out there.
